### ISSUE TITLE
about real dataset (labeled or unlabeled)

### ISSUE THREAD

**Issue by lx7555** [WangWenhao0716/DomainMix#4]
Thanks for your work, I noticed that you validated the importance of using unlabeled real datasets. 
So how do I control in code to implement data states ((labeled or unlabeled)) for training.

Thank you!

**Comment by WangWenhao0716** [comment]
They should be controlled by `train.py`:
```
train_loader = get_train_loader(args, dataset_real, args.height, args.width,
                                            args.batch_size, args.workers, args.num_instances, iters,
                                            trainset=pseudo_labeled_dataset_relabel + dataset_syne.train)
```
`pseudo_labeled_dataset_relabel` denotes using the unlabeled dataset, and `dataset_syne.train` denotes using the labels.

**Comment by lx7555** [comment]
Thank you. As your suggestion,

train_loader = get_train_loader(args, dataset_real, args.height, args.width,
                                            args.batch_size, args.workers, args.num_instances, iters,
                                            trainset=pseudo_labeled_dataset_relabel) 

 mean real data using the unlabeled dataset (pseudo_labeled) 

while
train_loader = get_train_loader(args, dataset_real, args.height, args.width,
                                            args.batch_size, args.workers, args.num_instances, iters,
                                            trainset=pseudo_labeled_dataset_relabel + dataset_syne.train)
 mean real data using the labeled dataset (real_labeled)

**Comment by WangWenhao0716** [comment]
No.
```
train_loader = get_train_loader(args, dataset_real, args.height, args.width,
args.batch_size, args.workers, args.num_instances, iters,
trainset=pseudo_labeled_dataset_relabel)
```
means real data using the unlabeled dataset (pseudo_labeled)

```
train_loader = get_train_loader(args, dataset_real, args.height, args.width,
args.batch_size, args.workers, args.num_instances, iters,
trainset=pseudo_labeled_dataset_relabel + dataset_syne.train)
```
means real data using the unlabeled dataset (pseudo_labeled) while syne data using the labeled dataset.

**Comment by lx7555** [comment]
Thank you!
So as you mentioned, how to use  "adding labeled real-world data to synthetic data" in your paper?
I mean how to change the label state for real-world data?
Best wishes

**Comment by lx7555** [comment]
Maybe?
train_loader = get_train_loader(args, dataset_real, args.height, args.width,
args.batch_size, args.workers, args.num_instances, iters,
trainset=dataset_real.train + dataset_syne.train

means real data using the labeled dataset (real-labeled) while syne data using the labeled dataset.

**Comment by WangWenhao0716** [comment]
Yes.

**Comment by lx7555** [comment]
> Yes.

Thank you
